plugins/tiddlywiki/tiddlyweb/syncer-actions-copy-logs.tid

```diff
@@ -0,0 +1,6 @@
+title: $:/plugins/tiddlywiki/tiddlyweb/syncer-actions/copy-logs
+tags: $:/tags/SyncerDropdown
+
+<$button message="tm-copy-syncer-logs-to-clipboard" class="tc-btn-invisible">
+{{$:/core/images/copy-clipboard}} Copy syncer logs to clipboard
```

Comment:
I'm OK with those "snippets" for readability ... Just a reminder, that we need to translate this texts ;)

Reply:
Yes indeed. We've got @sycom's #4179 which could address it. Equally, I still wonder whether #1011 might be a good idea.

Reply:
about: #1011 ... I think these are 2 different elements of a server side middleware. `tiddlyweb` contains the tiddlyweb API abstraction for the client. filesystem contains the "storage API" below tiddlyweb api.

So it should be possible to exchange filesystem with an sql-like adapter and it should still work. The same may be true for a tiddlyweb / dat-fs combination.

What I want to say is, that I'm not convinced any-more, that the 2 elements should be combined. 

The problem is, that we would need much clearer API definitions to make it easier for devs to actually implement different types of middleware.

Reply:
For clarity, the tiddlyweb and filesystem plugins are both instances of a "syncadaptor". The syncer that runs on the server uses the filesystem adaptor, and the syncer that runs in the browser uses the tiddlyweb adaptor. It is certainly possible to replace either syncadaptor with a different one.

If we integrated the tiddlyweb and filesystem plugins in the core we'd need some way to disable them so that users could choose to use different syncadaptors.